packages/contracts/contracts/optimistic-ethereum/OVM/bridge/messaging/OVM_L1CrossDomainMessenger.sol

```diff
@@ -114,6 +167,11 @@ contract OVM_L1CrossDomainMessenger is iOVM_L1CrossDomainMessenger, Abs_BaseCros
             "Provided message has already been received."
         );
 
+        require(
+            blockedMessages[xDomainCalldataHash] == false,
```

Comment:
SLOAD on every call to `relayMessage()`

Reply:
I think this is fine for now, especially since we have a separate ticket to improve gas.  @elenadimitrova in terms of ways to lower gas usage in the future we could have a
```
struct messageState {
   blocked: bool,
   successful: bool
}
```
And use a single mapping, but I am guessing that even that is not a meaningful gas impact for the time being.